### ISSUE TITLE
SixTrack style guide

### ISSUE THREAD

**Issue by vkbo** [SixTrack/SixTrack#1051]
Have discussed this a few times before with @rdemaria, and as far as I remember we discussed using some adapted form of Python PEP8. It may be useful to type one up and add the few extra conventions that we've implemented via consensus.

A few other suggestions from me as well as conventions already used in parts of the code by others:
* Indentation is two spaces. No tabs anywhere.
* Module content (routine/function declarations) should start on the first character. No initial indentation.
* Module routines and variables should be prefixed by a keyword and underscore to create a simple name space. (@kyrsjo)
* Use statements in module headers should be limited to the absolute minimum. @kyrsjo has argued this point well in the past pointing out that the content of included modules bleed throug to where the parent module is used as well. I have generally tried to limit this to the constants we use for initialisation, as well as floatPrecision, in the more recent code.
* Module variables should be private by default.
* We no longer need compatibility with Fortran 77, so F77 style syntax should be avoided even if it isn't technically deprecated according to the standard. I'm especially thinking of the F77 style logical operators `.eq.`, `.ne.`, `.gt.`, `.lt.`, `.ge.`, and `.le.`. Instead `==`, `/=`, `>`, `<`, `>=`, and `<=` should be used.
* Code line length should be no more than 120 characters. My main argument for this is that it fits in the GitHub code area.
* Text blobs should be maximum 100 wide (exception for inline comments).

**Comment by JamesMolson** [comment]
Isn't the standard 132 chars?

**Comment by JamesMolson** [comment]
This is what I've always been using and from what I recall was agreed on when we did the f77 -> f90 switch

**Comment by vkbo** [comment]
The maximum allowed length of a code line by the standard doesn't really dictate coding style recommendations. The latter is for readability and flow. I personally disagree with the 80 character philosophy often used because it is too restrictive and makes the code very hard to read, but 100-120 range is sensible in my opinion. My only argument against longer lines is, as I mentioned, readability on GitHub.

In any case, my point was mostly to sum up what has been the more or less unwritten rules as well as recommendations I've found online. It is mostly conventions established by @kyrsjo, and from discussions with you (indentation) and @rdemaria (line length). Since tomorrow is my official last day, I don't think my opinion should carry that much weight, but I do think there should be some written guidelines to ensure the code is easier to maintain in the future. Especially since there are so many collaborators.

**Comment by rdemaria** [comment]
I also agree a style guide would be useful. Thanks Veronika for basically drafting it!

Ideally I would love to have a Fortran linter, because it is very difficult to maintain style. Has somebody tried this https://github.com/cphyc/fortran-syntax ?
